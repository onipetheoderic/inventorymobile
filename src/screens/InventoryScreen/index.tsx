import React, { useState, useEffect } from "react";

import {
  ScrollView,
  View,
  StyleSheet,
  TouchableOpacity,
  Text,
} from "react-native";
import { Ionicons, Entypo } from "@expo/vector-icons";
import { Header, SingleItems } from "../../components/common";
import AsyncStorage from "@react-native-community/async-storage";
import { useIsFocused } from "@react-navigation/native";

export default function InventoryScreen(props: any) {
  const [defaultItems, setDefaultItems] = useState<any[]>([]);
  const isFocused = useIsFocused();
  useEffect(() => {
    (async () => {
      try {
        let items = await AsyncStorage.getItem("@inventory");
        if (items != null) {
          const parsedItem = JSON.parse(items);
          setDefaultItems(parsedItem);
        }
      } catch (e) {}
    })();
  }, [props.navigation, isFocused]);

  return (
    <View>
      <Header navigation={props.navigation} headerText="Inventory Listing" />

      <ScrollView>
        <View style={styles.listingContainer}>
          {defaultItems.map(item => (
            <SingleItems
              key={item.id}
              navigation={props.navigation}
              name={item.name}
              unit_price={item.unit_price}
              quantity={item.quantity}
              currency={item.currency}
              category={item.category}
            />
          ))}
        </View>
      </ScrollView>
      <TouchableOpacity
        onPress={() => props.navigation.navigate("CreateInventory")}
        style={styles.addButton}
      >
        <Entypo name="plus" size={32} color="white" />
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  listingContainer: {
    width: "90%",
    minHeight: 200,
    alignSelf: "center",
    justifyContent: "space-between",
    flexDirection: "row",
    flexWrap: "wrap",
    marginBottom: 120,
  },
  addButton: {
    alignItems: "center",
    justifyContent: "center",
    position: "absolute",
    backgroundColor: "blue",
    width: 60,
    height: 60,
    borderRadius: 30,
    top: 400,
    right: 20,
    zIndex: 400,
  },
});
