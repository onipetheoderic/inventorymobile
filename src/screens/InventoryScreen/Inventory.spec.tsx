import React from "react";
jest.mock("@react-native-community/async-storage", () =>
  require("@react-native-community/async-storage/jest/async-storage-mock")
);
jest.mock("@react-navigation/native");
import renderer from "react-test-renderer";

import InventoryScreen from "../InventoryScreen/index";

describe("<InventoryScreen />", () => {
  it("matches last snapshot", async () => {
    const tree = renderer.create(<InventoryScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
