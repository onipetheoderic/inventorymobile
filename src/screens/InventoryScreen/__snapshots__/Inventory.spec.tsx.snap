// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InventoryScreen /> matches last snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "height": 100,
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginLeft": 15,
          "opacity": 1,
        }
      }
      testID="back-button"
    >
      <Text />
    </View>
    <Text
      style={
        Object {
          "flex": 3,
          "fontFamily": "ReemKufi_400Regular",
          "fontSize": 19,
          "textAlign": "center",
        }
      }
    >
      Inventory Listing
    </Text>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 30,
            "elevation": 12,
            "height": 60,
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": 60,
            "zIndex": 10,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-between",
            "marginBottom": 120,
            "minHeight": 200,
            "width": "90%",
          }
        }
      />
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "blue",
        "borderRadius": 30,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 20,
        "top": 400,
        "width": 60,
        "zIndex": 400,
      }
    }
  >
    <Text />
  </View>
</View>
`;
